ghost/core/core/server/data/migrations/versions/6.0/2025-06-24-09-19-42-use-object-id-for-owner-user.js

```diff
@@ -0,0 +1,167 @@
+            return;
+        }
+
+        const email = currentOwnerUser.email;
+        const slug = currentOwnerUser.slug;
+        const temporaryEmail = `migrating-${newId}`;
+        const temporarySlug = `migrating-${newId}`;
+
+        // Step 1: Create a clone of the current owner user using the new ID.
+        // This is so we can assign existing references to the new owner user ID
+        // without violating foreign key constraints. We also need to use a
+        // temporary email and slug to avoid duplicate constraint errors
+        const clonedOwnerUser = {
+            ...currentOwnerUser,
+            id: newId,
+            slug: temporarySlug,
+            email: temporaryEmail
+        };
+
+        await knex('users').insert(clonedOwnerUser);
+
+        logging.info(`Cloned existing owner user with updated properties - ID: ${newId}, email: ${temporaryEmail}, slug: ${temporarySlug}`);
+
+        // Step 2: Update all references to the current owner user ID to point
+        // to the cloned owner user ID. The order matters to avoid foreign key
+        // constraint violations.
+
+        // 1. Tables with user_id column
+        await knex('roles_users').where('user_id', LEGACY_OWNER_USER_ID).update({user_id: newId});
+        await knex('permissions_users').where('user_id', LEGACY_OWNER_USER_ID).update({user_id: newId});
+        await knex('sessions').where('user_id', LEGACY_OWNER_USER_ID).update({user_id: newId});
+        await knex('api_keys').where('user_id', LEGACY_OWNER_USER_ID).update({user_id: newId});
+
+        // 2. Tables with author_id column
+        await knex('posts_authors').where('author_id', LEGACY_OWNER_USER_ID).update({author_id: newId});
+        await knex('post_revisions').where('author_id', LEGACY_OWNER_USER_ID).update({author_id: newId});
+
+        // 3. Tables with published_by column (nullable)
+        await knex('posts').where('published_by', LEGACY_OWNER_USER_ID).update({published_by: newId});
+
+        // 4. Actions table with actor_id
+        await knex('actions')
+            .where('actor_id', LEGACY_OWNER_USER_ID)
+            .where('actor_type', 'user')
+            .update({actor_id: newId});
+
+        // Step 3: Clean up the now redundant user record identified by the
+        // legacy owner user ID as there should be no references to it anymore
+        await knex('users').where('id', LEGACY_OWNER_USER_ID).del();
+
+        logging.info(`Removed user with ID: ${LEGACY_OWNER_USER_ID}`);
+
+        // Step 4: Restore the original slug and email on the cloned owner user
+        // record
+        await knex('users').where('id', newId).update({
+            slug,
+            email
+        });
+
+        logging.info(`Restored owner user slug and email: ${slug}, ${email}`);
+
+        logging.info(`Successfully updated owner user ID from ${LEGACY_OWNER_USER_ID} to ${newId}`);
+    },
+
+    async function down(knex) {
```

Comment:
Since this is being done as part of a major version, do we even need this? (i.e we can't rollback to the previous state anyway)

Reply:
No, there's no expectation that we have a down function